server.py

```diff
@@ -65,215 +57,364 @@ def create_test_room():
+    targetId = request.form.get('targetId')
+    try:
+        room = Room.objects.get(roomId=roomId)
+        killer = room.active_mafia(userId)
+        if not killer:
+            return build_response({"message": "Not mafia"}, 400)
+        name = killer.name
+        target_player = room.get_player(targetId)
+        if target_player.status == 'alive' and killer.status == 'alive':
+            room.targets.killTarget = target_player.userId
+            room.phase = 'doctor'
+            room.gameMessages.append(GameMessage(
+                primary='Doctor Phase', secondary='Doctor pick someone to heal'))
+            room.observerMessages.append(ObserverMessage(
+                primary='Mafia Action', secondary=name + ' killed' + target_player.name))
+            room.save()
+            return build_response({"message": "Target confirmed"}, 200)
+        else:
+            return build_response({"message": "Not valid target"}, 400)
+    except Exception as e:
+        print(e)
+        return build_response({"message": "Unexpected server error"}, 500)
+
+
+@ app.route('/rooms/<roomId>/heal', methods=['PATCH', 'OPTIONS'])
+@cross_origin(supports_credentials=True)
 def doctor_actions(roomId):
     LOG.info(request.access_route[0] + ' requested ' + request.url)
-    if request.method == 'OPTIONS':
-        return build_preflight_response()
-    elif request.method == 'PATCH':
-        userId = request.cookies.get('userId')
-        is_doctor, name = check_doctor(database, roomId, userId)
-        if not is_doctor:
-            return build_actual_response({"message": "Not doctor"}, 400)
-        is_valid_target = heal_action(
-            database, roomId, name, request.form.get('targetId'))
-        if not is_valid_target:
-            return build_actual_response({"message": "Not valid target"}, 400)
-        return build_actual_response({"message": "Target confirmed"}, 200)
-
-
-@app.route('/rooms/<roomId>/check', methods=['PATCH', 'OPTIONS'])
+    userId = request.cookies.get('userId')
+    targetId = request.form.get('targetId')
+    try:
+        room = Room.objects.get(roomId=roomId)
+        doctor = room.doctor
+        name = doctor.name
+        if userId != doctor.userId:
+            return build_response({"message": "Not doctor"}, 400)
+        target_player = room.get_player(targetId)
+        if target_player.status == 'alive' and doctor.status == 'alive':
+            room.targets.healTarget = target_player.userId
+            room.phase = 'detective'
+            room.gameMessages.append(
+                GameMessage(primary='Detective Phase', secondary='Detective pick someone to check'))
+            room.observerMessages.append(
+                ObserverMessage(primary='Doctor Action', secondary=name + " healed " + target_player.name))
+            room.save()
+            return build_response({"message": "Target confirmed"}, 200)
+        else:
+            return build_response({"message": "Not valid target"}, 400)
```

Comment:
so you never get into a state where `if target_player.status == 'alive' and doctor.status != 'alive':` because the game will skip their turn, right? or for now the room master has to know to skip turn?

Reply:
yeah I assume that the room master takes care of the turn skipping